docs/llm_tools.md

```diff
@@ -0,0 +1,204 @@
+---
+title: "LLM Tools"
+sidebar_label: "LLM Tools"
+---
+
+LLM tools extend the functionality of OpenAI Conversation, Google Generative AI, and possibly other future LLM integrations. You can define your own LLM tool that these integrations can use.
+
+There are 3 options to define your own tool.
+
+## Option 1. Intent
+
+Create and register your own intent handler. Intents are automatically registered as LLM tools. This method allows to use the same code for other integrations that support intents.
+
+## Option 2. Tool class
+
+Define a child class of `homeassistant.helpers.llm.Tool` and register an object of that class. This it the most flexible method. Example:
+
+```python
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import llm
+
+
+class MyTool(llm.Tool):
+    """LLM multiplication helper tool."""
+
+    name = "multiply"
+    description = "Return a product of two integers"
+    parameters = vol.Schema(
+        {
+            vol.Required("a", description="multiplicand"): vol.Coerce(int),
+            vol.Required("b", description="multiplier"): vol.Coerce(int),
+        }
+    )
+
+    async def async_call(self, hass: HomeAssistant, tool_input: llm.ToolInput) -> Any:
+        """Call the function."""
+        args = tool_input.tool_args
+        return {"result": args["a"] * args["b"]}
+
+
+my_tool = MyTool()
+llm.async_register_tool(hass, my_tool)
+```
+
+Let's unpack.
+
+The `llm.Tool` class has the following user-defined attributes:
+
+| Name                | Type       | Description                                                                                                    |
+|---------------------|------------|----------------------------------------------------------------------------------------------------------------|
+| `name`              | string     | The name of the tool. Required.                                                                                |
+| `description`       | string     | Description of the tool to help the LLM understand when and how it should be called. Optional but recommended. |
+| `parameters`        | vol.Schema | The voluptuous schema of the parameters. Defaults to vol.Schema({})                                            |
+
+The `llm.Tool` class has the following user-defined methods:
+
+#### `async_call`
+Perform the actual operation of the tool when called by the LLM. This must be an async method and return json-serializable result. Its arguments are `hass` and an instance of `llm.ToolInput`.
```

Comment:
I think it would be good to have similar to https://developers.home-assistant.io/docs/dev_101_services/#response-data (1) a more specific definition of the response type and (2)  More specific definition of how errors are handled.

Below you have "json-serializable result." which is good and so that offers a way to be more specific here.